sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ServiceTag.java

```diff
@@ -0,0 +1,136 @@
+ * Use service tags in place of specific IP addresses when you create security rules and routes.
+ */
+public class ServiceTag {
+    private static final Map<String, ServiceTag> VALUES_BY_NAME = new ConcurrentHashMap<>();
+    /** Action Group. */
+    public static final ServiceTag ACTION_GROUP = new ServiceTag("ActionGroup", true, false, false, false);
+    /** Azure Storage.
+     * <p>Note: This tag represents the service, but not specific instances of the service.
+     * For example, the tag represents the Azure Storage service, but not a specific Azure Storage account.</p>
+     */
+    public static final ServiceTag STORAGE = new ServiceTag("Storage", false, true, true, true);
+    /**
+     * The IP address space that's outside the virtual network and reachable by the public internet.
+     * The address range includes the <a href="https://www.microsoft.com/en-us/download/details.aspx?id=41653">Azure-owned public IP address space</a>.
+     */
+    public static final ServiceTag INTERNET = new ServiceTag("Internet", true, true, false, false);
+    /**
+     * The virtual network address space (all IP address ranges defined for the virtual network),
+     * all connected on-premises address spaces, peered virtual networks, virtual networks connected to a virtual network gateway,
+     * the virtual IP address of the host, and address prefixes used on user-defined routes.
+     * This tag might also contain default routes.
+     */
+    public static final ServiceTag VIRTUAL_NETWORK = new ServiceTag("VirtualNetwork", true, false, false, false);
+
+    /**
+     * Get service tag from name.
+     *
+     * @param serviceTagName the name of the service tag, can be with or without region
+     * @return service tag, null if invalid
+     */
+    public static ServiceTag fromName(String serviceTagName) {
+        if (serviceTagName == null) {
+            return null;
+        }
+        String[] tagParts = serviceTagName.split(Pattern.quote("."));
+        if (tagParts.length == 1) { // root Tag
+            if (serviceTagName.contains(".")) {
+                return null;
+            }
+            return VALUES_BY_NAME.get(tagParts[0]);
+        } else if (tagParts.length == 2) { // Tag.Region
+            ServiceTag rootTag = VALUES_BY_NAME.get(tagParts[0]);
+            if (rootTag == null || CoreUtils.isNullOrEmpty(tagParts[1])) {
+                return null;
+            }
+            return rootTag.withRegion(Region.fromName(tagParts[1]));
+        } else { // unknown
+            return null;
+        }
+    }
+
+    /**
+     * Get regional service tag.
+     *
+     * @param region the region of the regional service tag, will be <strong>omitted</strong> if
+     *               <p>It is null.</p>
+     *               <p>The root service tag either cannot be regional or is already regional.</p>
+     * @return regional service tag
+     */
+    public ServiceTag withRegion(Region region) {
```

Comment:
should we have a subclass for ServiceTag + Region?

also not sure about the name, it looks like a setter method, but then return a new object.

PS: `ServiceTag` should be immutable.

Reply:
I agree, we can add a subclass to have `region` as its field and override the `name()` method.

As with the name, what about `regional()` or `regionalTag`?

Reply:
Or e.g. `RegionalServiceTag.fromServiceTag()`?

Reply:
I have one concern. As we are taking `ServiceTag` as setter argument, how does the user know about this additional `RegionalServiceTag` that they can use?

Reply:
They need to know `RegionalServiceTag` first (somehow).

Or you can have `ServiceTag.toRegional`?

Reply:
I'm more on `ServiceTag.toRegional`. We can expect `RegionalServiceTag.toRegional` to do the same thing, which is 
to return a new `RegionalServiceTag` with regional part replaced.